bokeh/models/mappers.py

```diff
@@ -63,3 +65,229 @@ class LinearColorMapper(ColorMapper):
     def __init__(self, palette=None, **kwargs):
         if palette is not None: kwargs['palette'] = palette
         super(LinearColorMapper, self).__init__(**kwargs)
+
+class SegmentedColorMapper(ColorMapper):
+    """ Map numbers in a range [*low*, *high*] linearly into a
+    sequence of colors which are defined by a seires of breaks
+    correlated to different colors.
+
+    For example, if the breaks are specified 0, 0.5, 1 and the
+    colors are defined as 'red', 'white', 'green', values 
+    between 0 and 0.5 will smoothy vary between red and white
+    while values between 0.5 and 1 will vary between white and
+    green.
+    """
+
+    palette = Seq(Color, help="""
+    A sequence of colors to use as the target palette for mapping.
+
+    This property can also be set as a ``String``, to the name of
+    any of the palettes shown in :ref:`bokeh_dot_palettes`.
+    """).accepts(Enum(Palette), lambda pal: getattr(palettes, pal))
+
+    segments = Seq(Float, help="""
+    A listing of values which are directly tied to the 
+    colors defined in the palette.  The list must have
+    at least two elements.  If there are exactly two
+    elements, then the resulting colormap will be
+    interpolated between these two values and across
+    all of the colors defined in the palette.  If there
+    are more than two colors, the length must match the
+    number of colors in the palette.  In this configuration
+    each segment value is tied to the palette color of the
+    same index.  This argument is optional.  If it is
+    not provided, the min and max of the data will be used.
+    """)
+
+    alpha = Seq(Float, help="""
+    A python array, the same length as the dictionary with the
+    desired alpha values
+    """
+    )
```

Comment:
Well, I think this is pretty handy here.  Named colors do not carry an alpha channel with them and those are the easiest to use for an end user.  While it is possible to provide the alpha information in  the tuple or the hex codes, who has time to look those up for lime green.  :)

This still leaves open the issue of what would be authoritative.  What if I specified a color like '#e3FF22aa' and then specified a single scalar alpha of 0.5?

Current behavior in the SegmentedColorMapper is to let the alpha values encoded in the palette be authoritative.  This seems like the right thing to do as it takes more effort to encode an alpha value there and that is most likely what the end user is going to want.